### PR TITLE
Create plugin for payments-example

### LINKED ISSUE #6129: move payments-example plugin to its own npm package, and install npm version Reaction
New repo: https://github.com/reactioncommerce/plugin-payments-example

Convert the payments-example plugin. This includes moving all related files into a new repository, releasing that repo as an `npm` package, and then updating this repo, `reaction`, to use the `npm` version of the plugin.

Part of [this epic](https://github.com/reactioncommerce/reaction/issues/6107) to move all plugins into their own repositories and release as `npm` packages.

### PR DESCRIPTION
**Pull request by trojanh**
Resolves https://github.com/reactioncommerce/reaction/issues/6129
Impact: **minor**
Type: **refactor**

## Issue
Convert the payments-example plugin. This includes moving all related files into a new repository, releasing that repo as an npm package, and then updating this repo, reaction, to use the npm version of the plugin.

Part of this epic to move all plugins into their own repositories and release as npm packages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+  "name": "@reactioncommerce/plugin-payments-example",
+  "description": "Example Payments plugin for the Reaction API",
+  "version": "1.0.0",
```
Comment (kieckhafer) [inline package.json]: Version should be changed to `0.0.0-development", `semantic-release` will take care of bumping the version numbers, it doesn't need to be set here.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * @summary Import and call this function to add this plugin to your API.
+ * @param {ReactionAPI} app The ReactionAPI instance
+ * @returns {undefined}
+ */
+export default async function register(app) {
+  await app.registerPlugin({
+    label: "ExamplePayment",
+    name: "example-paymentmethod",
+    version: app.context.appVersion,
```
Comment (kieckhafer) [inline src/index.js]: This version should be imported from `package.json`.

Change this line to `version: pkg.version`, and then `import pkg from "../package.json";`

**Review by kieckhafer — CHANGES_REQUESTED** [review]
A couple small changes just looking at the code. Will test installed next.

**Comment by kieckhafer** [comment]
@trojanh looks great overall, a couple small comments, but works as expected aside from those. Will merge once those comments addressed.

**Comment by trojanh** [comment]
great, thanks for reviewing @kieckhafer , this new way of importing plugins from the package is really cool and keeps the main API code clean. 

If I get a chance I'll move more such plugins to npm packages.

**Comment by trojanh** [comment]
I have updated the code as per your suggestion.

**Review by kieckhafer — APPROVED** [review]

**Pull request merged**